src/cbc_sdk/base.py

```diff
@@ -576,19 +870,50 @@ def _perform_query(self):
         yield from ()
 
     def __len__(self):
+        """
+        Returns the number of items returned by this query.
+
+        Returns:
+            int: The number of items returned by this query.
+        """
         return 0
 
     def __getitem__(self, item):
+        """
+        Return a specific item or items from the query.
+
+        Args:
+            item (object): Indicates the item(s) to retrieve, either as an int or a slice.
+
+        Returns:
+            object: Either an item or a list of items.
+        """
         return None
 
     def __iter__(self):
+        """
+        Returns an iterator over the items returned by this query.
+
+        Returns:
+            Iterator: An iterator over the items returned by this query.
+        """
         return self._perform_query()
 
 
 class SimpleQuery(BaseQuery):
+    """A simple query object."""
     _multiple_where_clauses_accepted = False
 
     def __init__(self, cls, cb, urlobject=None, returns_fulldoc=True):
+        """
+        Initialize the SimpleQuery object.
+
+        Args:
+            cls (class): Class of the object to be returned by the query.
+            cb (CBCloudAPI): Reference to the CBCloudAPI object.
+            urlobject (str): URL to be used in making the query.
+            returns_fulldoc (bool): TBD
```

Comment:
`returns_fulldoc` eventually becomes `full_doc` in `SimpleQuery.results`. `full_doc` becomes `NewBaseModel._full_init` in `NewBaseModel.__init__()`. This is a flag for if the object has been fully initialized with data from CB. So for the description of `returns_fulldoc`, I would put "Whether the result of the Query yields objects that have been fully initialized", or something along those lines

Reply:
I'll use just what you say.